File: apps/kratos-web/src/enums/LinkFormat.ts
const LinkFormat = {
    GIF: 'gif',
    JPEG: 'jpeg',
    MOV: 'mov',
    MP3: 'mp3',
    MP4: 'mp4',
    PNG: 'png',
    SVG: 'svg',
} as const;

type LinkFormat = typeof LinkFormat[keyof typeof LinkFormat];

export default LinkFormat;

File: apis/kratos-api/src/enums/link_format.rs
use database::impl_bindable;
use serde::{Deserialize, Serialize};
use sqlx::Type;
use strum_macros::Display;

#[derive(Clone, Debug, Deserialize, Display, PartialEq, Serialize, Type)]
#[serde(rename_all = "snake_case")]
#[sqlx(rename_all = "snake_case", type_name = "varchar")]
#[strum(serialize_all = "snake_case")]
pub enum LinkFormat {
    Gif,
    Html,
    Jpeg,
    Mov,
    Mp3,
    Mp4,
    Png,
    Svg,
}

impl_bindable!(LinkFormat);

impl Default for LinkFormat {
    fn default() -> Self {
        Self::Jpeg
    }
}
